### PR TITLE
Step 6 - Adding filters

### PR DESCRIPTION
**Pull request by pablomarinoacn**

### CODE REVIEW

File: app/src/main/java/com/mstudio/superheroarch/presentation/overview/MainActivity.kt [hunk 1]
```diff
@@ -31,19 +31,34 @@ class MainActivity : AppCompatActivity() {
...
+            }
         }
     }
 
-    private fun getCharacters() {
+    private fun getCharacters(filter: String = StatusFilters.ALL.status) {
         val apiService = RetrofitInstance.retrofit().create(RickAndMortyApi::class.java)
         CoroutineScope(Dispatchers.IO).launch {
-            val result = apiService.getCharacters()
+            val result = apiService.getCharacters(filter)
```
Comment (PabloGrelaM) [inline app/src/main/java/com/mstudio/superheroarch/presentation/overview/MainActivity.kt:61]: There is no need of making a call every time a filter is pressed. You are retrieving all characters at the beginning, use that list to work with filters.

**Review by PabloGrelaM — CHANGES_REQUESTED** [review]
check comment

File: app/src/main/java/com/mstudio/superheroarch/presentation/overview/CharactersAdapter.kt [hunk 2]
```diff
@@ -13,9 +14,10 @@ class CharactersAdapter(
 
     private var items = listOf<CharactersRemoteEntity>()
 
+    @SuppressLint("NotifyDataSetChanged")
     fun updateItems(dataList: List<CharactersRemoteEntity>) {
         items = dataList
-        notifyItemRangeChanged(0, items.size)
+        notifyDataSetChanged()
```
Comment (PabloGrelaM) [inline app/src/main/java/com/mstudio/superheroarch/presentation/overview/CharactersAdapter.kt:20]: still deprecated

File: app/src/main/java/com/mstudio/superheroarch/presentation/overview/MainActivity.kt [hunk 3]
```diff
@@ -25,18 +25,34 @@ class MainActivity : AppCompatActivity() {
...
+            aliveFilterButton.text = resources.getString(R.string.alive_filter_button_title)
+            deadFilterButton.text = resources.getString(R.string.dead_filter_button_title)
+            unknownFilterButton.text = resources.getString(R.string.unknown_filter_button_title)
+            allFilterButton.text = resources.getString(R.string.all_filter_button_title)
+            aliveFilterButton.setOnClickListener {
+                adapter.updateItems(allCharacters.filter { it.status.equals(StatusFilters.ALIVE.status, ignoreCase = true) })
             }
 
+            deadFilterButton.setOnClickListener {
+                adapter.updateItems(allCharacters.filter { it.status.equals(StatusFilters.DEAD.status, ignoreCase = true) })
```
Comment (PabloGrelaM) [inline app/src/main/java/com/mstudio/superheroarch/presentation/overview/MainActivity.kt:46]: this can be moved to a method.

**Review by PabloGrelaM — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/mstudio/superheroarch/presentation/overview/CharactersAdapter.kt (hunk 2 above)
Comment (pablomarinoacn) [reply app/src/main/java/com/mstudio/superheroarch/presentation/overview/CharactersAdapter.kt:20]: with notifyitemrangechanged is not working ... for this I decided to use this again 🤔

**Review by PabloGrelaM — APPROVED** [review]

**Pull request merged**